src/Hamiltonians/Molecular.jl

```diff
@@ -0,0 +1,388 @@
+end
+
+function operator_column(h::MolecularHamiltonian{T,A,D}, a::A)::MolecularHamiltonianOperatorColumn where {T<:Number,A<:FermiFS2C,D}
+    modes = modes_extract(a)
+
+    diag = zero(T)
+    diag_one_elec_int = one_electron_integral(h.fcidump.int1, modes.occupied)
+    diag_two_elec_int = two_electron_integral(h.fcidump.int2, modes.occupied)
+    diag = h.fcidump.int0 + diag_one_elec_int + diag_two_elec_int
+
+    MolecularHamiltonianOperatorColumn{A,T,typeof(h),typeof(modes)}(a, h, diag, modes)
+end
+
+parent_operator(c::MolecularHamiltonianOperatorColumn) = c.op
+starting_address(c::MolecularHamiltonianOperatorColumn) = c.address
+
+function diagonal_element(column::MolecularHamiltonianOperatorColumn{A,T,O,OD}) where {A<:FermiFS2C,T,O,OD}
+    return column.diag
+end
+
+function num_offdiagonals(column::MolecularHamiltonianOperatorColumn)
+    return length(offdiagonals(column))
+end
+
+function offdiagonals(column::MolecularHamiltonianOperatorColumn)
+    return Iterators.flatten(get_offdiagonal_iterators(column.address, column.op, column.modes))
+end
+
+function random_offdiagonal(column::MolecularHamiltonianOperatorColumn)
+    ods = offdiagonals(column)
+    r = rand(collect(ods))
+    return r[1], 1 / length(ods), r[2]
+end
+
+# function random_offdiagonal(column::MolecularHamiltonianOperatorColumn)
+#     return rand(column.ods), 1 / length(column.ods)
+# end
+
+function one_electron_integral(int1::Array{T,2}, occ_modes::Tuple{AbstractVector{FermiFSIndex},AbstractVector{FermiFSIndex}}) where {T<:Number}
+    one_elec_int = zero(T)
+    for occ_mode in occ_modes
+        for i in occ_mode
+            one_elec_int += int1[i.mode, i.mode]
+        end
+    end
+    one_elec_int
+end
+
+function two_electron_integral(int2::Array{T,4}, occ_modes::Tuple{AbstractVector{FermiFSIndex},AbstractVector{FermiFSIndex}})::T where {T<:Number}
+    two_elec_int = zero(T)
+
+    sum_alpha_alpha = zero(T)
+    for i in occ_modes[1]
+        for j in occ_modes[1]
+            if i.mode ≠ j.mode
+                sum_alpha_alpha += int2[i.mode, j.mode, i.mode, j.mode] - int2[i.mode, j.mode, j.mode, i.mode]
+            end
+        end
+    end
+    two_elec_int += 0.5 * sum_alpha_alpha
+
+    sum_alpha_beta = zero(T)
+    for i in occ_modes[1]
+        for j in occ_modes[2]
+            sum_alpha_beta += int2[i.mode, j.mode, i.mode, j.mode]
+        end
+    end
+    two_elec_int += sum_alpha_beta
+
+    sum_beta_beta = zero(T)
+    for i in occ_modes[2]
+        for j in occ_modes[2]
+            if i.mode ≠ j.mode
+                sum_beta_beta += int2[i.mode, j.mode, i.mode, j.mode] - int2[i.mode, j.mode, j.mode, i.mode]
+            end
+        end
+    end
+    two_elec_int += 0.5 * sum_beta_beta
+
+    two_elec_int
+end
+
+@inline function flip_spin_chan(chan::Int)::Int
+    if chan == 1
+        return 2
+    else
+        chan == 2
+        return 1
+    end
+end
+
+struct ModeIndex{N}
+    index::NTuple{N,Int64}
+end
```

Comment:
I don't understand what `ModeIndex` is doing or what it is good for. Please add docstrings also for internal types or functions to help the reader (unless your code is very simple and self-documenting).